Cargo.toml

```diff
@@ -34,6 +34,7 @@ paste = "1.0.14"
 raw-cpuid = "11.0.1"
 strum = { version = "0.25.0", features = ["derive"] }
 to_method = "1.1.0"
+try-lock = "0.2.5"
```

Comment:
As discussed offline, I don't think we should add this dependency in this PR. I don't want to block this PR while we consider the new dependency, and it would be better to entirely switch over if we choose to use it.

Reply:
I'll replace it with a `Mutex` for now and then merge.